app/src/main/java/org/p2p/wallet/history/ui/token/adapter/HistoryAdapter.kt

```diff
@@ -99,21 +102,29 @@ class HistoryAdapter(
         return currentItems.size + additionalItemSize
     }
 
-    private fun getTransactionItemViewType(position: Int): Int {
-        return when (currentItems[position]) {
+    private fun getHistoryItemViewType(position: Int): Int {
+        return when (val item = currentItems[position]) {
             is HistoryItem.DateItem -> HISTORY_DATE_VIEW_TYPE
-            is HistoryItem.TransactionItem -> TRANSACTION_VIEW_TYPE
+            is HistoryItem.TransactionItem -> getTransactionItemViewType(item)
             is HistoryItem.Empty -> HISTORY_EMPTY_VIEW_TYPE
         }
     }
 
+    private fun getTransactionItemViewType(item: HistoryItem.TransactionItem): Int {
+        return when (item.transaction) {
+            is HistoryTransaction.Swap -> TRANSACTION_SWAP_VIEW_TYPE
+            else -> TRANSACTION_VIEW_TYPE
```

Comment:
Are send and receive has one view type? The naming is not clear. One is `TRANSACTION_SWAP_VIEW_TYPE` and other is `TRANSACTION_(?)_VIEW_TYPE`